### ISSUE TITLE
Install process doesn't properly build `lttbc`

### ISSUE THREAD

**Issue by darynwhite** [conda-forge/plotly-resampler-feedstock#30]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

I have seen the `UserWarning` for the slower, Python version of `lttbc` and finally determined that it is likely caused by when I install via `conda-forge` as this recipe doesn't contain any parameters to compile the internal C code.

This happens to coincide with the version of `lttbc` that is on `conda-forge` does not have the same functions or structure that is relied upon by `plotly-resampler`. 

I have tested this by building an environment that installed `plotly-resampler` via `pip` and the `UserWarning` never appeared.

### Installed packages

```shell
(resampler-test) ❯ conda list
# packages in environment at /Users/white/conda/envs/resampler-test:
#
# Name                    Version                   Build  Channel

(resampler-test) ❯ pip list
Package                       Version
----------------------------- -----------
aiofiles                      22.1.0
aiosqlite                     0.18.0
ansi2html                     1.8.0
anyio                         3.6.2
appnope                       0.1.3
argon2-cffi                   21.3.0
argon2-cffi-bindings          21.2.0
asttokens                     2.2.1
attrs                         22.2.0
Babel                         2.12.1
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
beautifulsoup4                4.12.0
black                         23.3.0
bleach                        6.0.0
boltons                       23.0.0
Brotli                        1.0.9
brotlipy                      0.7.0
Cartopy                       0.21.1
certifi                       2022.12.7
cffi                          1.15.1
cftime                        1.6.2
charset-normalizer            2.1.1
click                         8.1.3
cmocean                       3.0.3
colorama                      0.4.6
colorlog                      6.7.0
colorspacious                 1.1.2
comm                          0.1.3
conda                         23.3.1
conda-package-handling        2.0.2
conda_package_streaming       0.7.0
conda-tree                    1.0.5
contourpy                     1.0.7
cryptography                  40.0.1
cycler                        0.11.0
dash                          2.9.2
dash-core-components          2.0.0
dash-html-components          2.0.0
dash-table                    5.0.0
debugpy                       1.6.6
decorator                     5.1.1
defusedxml                    0.7.1
entrypoints                   0.4
et-xmlfile                    1.1.0
executing                     1.2.0
fastjsonschema                2.16.3
Flask                         2.2.3
Flask-Compress                1.13
flit_core                     3.8.0
fonttools                     4.39.3
gsw                           3.6.16
idna                          3.4
importlib-metadata            6.1.0
importlib-resources           5.12.0
ipykernel                     6.22.0
ipyparallel                   8.5.1
ipython                       8.12.0
ipython-genutils              0.2.0
ipywidgets                    8.0.6
itsdangerous                  2.1.2
jedi                          0.18.2
Jinja2                        3.1.2
json5                         0.9.5
jsonpatch                     1.32
jsonpointer                   2.0
jsonschema                    4.17.3
jupyter                       1.0.0
jupyter_client                8.1.0
jupyter-console               6.6.3
jupyter_core                  5.3.0
jupyter-dash                  0.4.2
jupyter-events                0.6.3
jupyter_server                2.5.0
jupyter_server_fileid         0.8.0
jupyter_server_terminals      0.4.4
jupyter_server_ydoc           0.8.0
jupyter-ydoc                  0.2.3
jupyterlab                    3.6.3
jupyterlab-pygments           0.2.2
jupyterlab_server             2.22.0
jupyterlab-widgets            3.0.7
kiwisolver                    1.4.4
MarkupSafe                    2.1.2
matplotlib                    3.7.1
matplotlib-inline             0.1.6
mistune                       2.0.5
munkres                       1.1.4
mypy-extensions               1.0.0
nbclassic                     0.5.3
nbclient                      0.7.2
nbconvert                     7.2.10
nbformat                      5.8.0
nest-asyncio                  1.5.6
netCDF4                       1.6.3
networkx                      3.0
notebook                      6.5.3
notebook_shim                 0.2.2
ntplib                        0.4.0
numpy                         1.24.2
openpyxl                      3.1.1
orjson                        3.8.9
packaging                     23.0
pandas                        1.5.3
pandocfilters                 1.5.0
parso                         0.8.3
pathspec                      0.11.1
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        9.4.0
pip                           23.0.1
pkgutil_resolve_name          1.3.10
platformdirs                  3.2.0
plotly                        5.14.0
plotly-resampler              0.8.3.2
pluggy                        1.0.0
pooch                         1.7.0
prometheus-client             0.16.0
prompt-toolkit                3.0.38
psutil                        5.9.4
ptyprocess                    0.7.0
pure-eval                     0.2.2
pycodestyle                   2.10.0
pycosat                       0.6.4
pycparser                     2.21
Pygments                      2.14.0
PyMySQL                       1.0.3
pyOpenSSL                     23.1.1
pyparsing                     3.0.9
PyPDF2                        2.11.1
pyproj                        3.5.0
pyrsistent                    0.19.3
pyserial                      3.5
pyshp                         2.3.1
PySocks                       1.7.1
python-dateutil               2.8.2
python-json-logger            2.0.7
pytz                          2023.3
PyYAML                        6.0
pyzmq                         25.0.2
requests                      2.28.2
retrying                      1.3.4
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
ruamel.yaml                   0.17.21
ruamel.yaml.clib              0.2.7
scipy                         1.10.1
seawater                      3.3.4
Send2Trash                    1.8.0
setuptools                    67.6.1
shapely                       2.0.1
six                           1.16.0
sniffio                       1.3.0
soupsieve                     2.3.2.post1
stack-data                    0.6.2
tabulate                      0.9.0
tenacity                      8.2.2
terminado                     0.17.1
tinycss2                      1.2.1
tomli                         2.0.1
toolz                         0.12.0
tornado                       6.2
tqdm                          4.65.0
trace-updater                 0.0.9
traitlets                     5.9.0
typing_extensions             4.5.0
unicodedata2                  15.0.0
urllib3                       1.26.15
wcwidth                       0.2.6
webencodings                  0.5.1
WebOb                         1.8.7
websocket-client              1.5.1
Werkzeug                      2.2.3
wheel                         0.40.0
widgetsnbextension            4.0.7
xarray                        2023.3.0
xlrd                          2.0.1
y-py                          0.5.5
ypy-websocket                 0.8.2
zipp                          3.15.0
zstandard                     0.19.0
```

### Environment info

```shell
(resampler-test) ❯ conda info

     active environment : resampler-test
    active env location : /Users/white/conda/envs/resampler-test
            shell level : 1
       user config file : /Users/white/.condarc
 populated config files : /Users/white/.condarc
          conda version : 23.3.1
    conda-build version : not installed
         python version : 3.9.16.final.0
       virtual packages : __archspec=1=arm64
                          __osx=12.6.4=0
                          __unix=0=0
       base environment : /Users/white/conda  (writable)
      conda av data dir : /Users/white/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-arm64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-arm64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/white/conda/pkgs
                          /Users/white/.conda/pkgs
       envs directories : /Users/white/conda/envs
                          /Users/white/.conda/envs
               platform : osx-arm64
             user-agent : conda/23.3.1 requests/2.28.2 CPython/3.9.16 Darwin/21.6.0 OSX/12.6.4
                UID:GID : 687:15
             netrc file : None
           offline mode : False
```

**Comment by thewchan** [comment]
@conda-forge/core Advice needed here... it seems that the correct protocol is then to figure out what's going on with the version of @conda-forge/lttbc that's not working right?

**Comment by ocefpaf** [comment]
It is probably lacking a build step. It would be nice to check upstream. Maybe all it is missing is to run https://github.com/predict-idlab/plotly-resampler/blob/v0.8.3.2/build.py before pip installing.

PS: while this seems to affect the conda-forge package, the package list above does not list the channel and they seem to be from defaults instead.

**Comment by darynwhite** [comment]
@ocefpaf here are the results of setting up the test env, I solely ran an install with `pip` for this test.

```
(resampler-test) ❯ pip install plotly-resampler
Collecting plotly-resampler
  Downloading plotly_resampler-0.8.3.2.tar.gz (46 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.4/46.4 kB 2.2 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: dash<3.0.0,>=2.2.0 in ./conda/envs/work/lib/python3.10/site-packages (from plotly-resampler) (2.9.2)
Requirement already satisfied: plotly<6.0.0,>=5.5.0 in ./conda/envs/work/lib/python3.10/site-packages (from plotly-resampler) (5.14.0)
Requirement already satisfied: numpy>=1.14 in ./conda/envs/work/lib/python3.10/site-packages (from plotly-resampler) (1.24.2)
Requirement already satisfied: pandas<2.0.0,>=1.3.5 in ./conda/envs/work/lib/python3.10/site-packages (from plotly-resampler) (1.5.3)
Collecting orjson<4.0.0,>=3.8.0
  Downloading orjson-3.8.9-cp310-cp310-macosx_10_9_x86_64.macosx_11_0_arm64.macosx_10_9_universal2.whl (488 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 488.6/488.6 kB 10.0 MB/s eta 0:00:00
Collecting trace-updater>=0.0.8
  Downloading trace_updater-0.0.9-py3-none-any.whl (185 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 185.1/185.1 kB 20.4 MB/s eta 0:00:00
Collecting jupyter-dash>=0.4.2
  Using cached jupyter_dash-0.4.2-py3-none-any.whl (23 kB)
Collecting dash-table==5.0.0
  Using cached dash_table-5.0.0-py3-none-any.whl (3.9 kB)
Requirement already satisfied: Flask>=1.0.4 in ./conda/envs/work/lib/python3.10/site-packages (from dash<3.0.0,>=2.2.0->plotly-resampler) (2.2.3)
Collecting dash-core-components==2.0.0
  Using cached dash_core_components-2.0.0-py3-none-any.whl (3.8 kB)
Collecting dash-html-components==2.0.0
  Using cached dash_html_components-2.0.0-py3-none-any.whl (4.1 kB)
Requirement already satisfied: ipython in ./conda/envs/work/lib/python3.10/site-packages (from jupyter-dash>=0.4.2->plotly-resampler) (8.12.0)
Requirement already satisfied: nest-asyncio in ./conda/envs/work/lib/python3.10/site-packages (from jupyter-dash>=0.4.2->plotly-resampler) (1.5.6)
Collecting ansi2html
  Using cached ansi2html-1.8.0-py3-none-any.whl (16 kB)
Requirement already satisfied: requests in ./conda/envs/work/lib/python3.10/site-packages (from jupyter-dash>=0.4.2->plotly-resampler) (2.28.2)
Collecting retrying
  Downloading retrying-1.3.4-py3-none-any.whl (11 kB)
Requirement already satisfied: ipykernel in ./conda/envs/work/lib/python3.10/site-packages (from jupyter-dash>=0.4.2->plotly-resampler) (6.22.0)
Requirement already satisfied: python-dateutil>=2.8.1 in ./conda/envs/work/lib/python3.10/site-packages (from pandas<2.0.0,>=1.3.5->plotly-resampler) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in ./conda/envs/work/lib/python3.10/site-packages (from pandas<2.0.0,>=1.3.5->plotly-resampler) (2023.3)
Requirement already satisfied: tenacity>=6.2.0 in ./conda/envs/work/lib/python3.10/site-packages (from plotly<6.0.0,>=5.5.0->plotly-resampler) (8.2.2)
Requirement already satisfied: packaging in ./conda/envs/work/lib/python3.10/site-packages (from plotly<6.0.0,>=5.5.0->plotly-resampler) (23.0)
Requirement already satisfied: itsdangerous>=2.0 in ./conda/envs/work/lib/python3.10/site-packages (from Flask>=1.0.4->dash<3.0.0,>=2.2.0->plotly-resampler) (2.1.2)
Requirement already satisfied: Jinja2>=3.0 in ./conda/envs/work/lib/python3.10/site-packages (from Flask>=1.0.4->dash<3.0.0,>=2.2.0->plotly-resampler) (3.1.2)
Requirement already satisfied: Werkzeug>=2.2.2 in ./conda/envs/work/lib/python3.10/site-packages (from Flask>=1.0.4->dash<3.0.0,>=2.2.0->plotly-resampler) (2.2.3)
Requirement already satisfied: click>=8.0 in ./conda/envs/work/lib/python3.10/site-packages (from Flask>=1.0.4->dash<3.0.0,>=2.2.0->plotly-resampler) (8.1.3)
Requirement already satisfied: six>=1.5 in ./conda/envs/work/lib/python3.10/site-packages (from python-dateutil>=2.8.1->pandas<2.0.0,>=1.3.5->plotly-resampler) (1.16.0)
Requirement already satisfied: jupyter-client>=6.1.12 in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (8.1.0)
Requirement already satisfied: debugpy>=1.6.5 in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (1.6.6)
Requirement already satisfied: matplotlib-inline>=0.1 in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (0.1.6)
Requirement already satisfied: traitlets>=5.4.0 in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (5.9.0)
Requirement already satisfied: tornado>=6.1 in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (6.2)
Requirement already satisfied: psutil in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (5.9.4)
Requirement already satisfied: comm>=0.1.1 in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (0.1.3)
Requirement already satisfied: jupyter-core!=5.0.*,>=4.12 in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (5.3.0)
Requirement already satisfied: pyzmq>=20 in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (25.0.2)
Requirement already satisfied: appnope in ./conda/envs/work/lib/python3.10/site-packages (from ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (0.1.3)
Requirement already satisfied: stack-data in ./conda/envs/work/lib/python3.10/site-packages (from ipython->jupyter-dash>=0.4.2->plotly-resampler) (0.6.2)
Requirement already satisfied: pexpect>4.3 in ./conda/envs/work/lib/python3.10/site-packages (from ipython->jupyter-dash>=0.4.2->plotly-resampler) (4.8.0)
Requirement already satisfied: pickleshare in ./conda/envs/work/lib/python3.10/site-packages (from ipython->jupyter-dash>=0.4.2->plotly-resampler) (0.7.5)
Requirement already satisfied: backcall in ./conda/envs/work/lib/python3.10/site-packages (from ipython->jupyter-dash>=0.4.2->plotly-resampler) (0.2.0)
Requirement already satisfied: prompt-toolkit!=3.0.37,<3.1.0,>=3.0.30 in ./conda/envs/work/lib/python3.10/site-packages (from ipython->jupyter-dash>=0.4.2->plotly-resampler) (3.0.38)
Requirement already satisfied: jedi>=0.16 in ./conda/envs/work/lib/python3.10/site-packages (from ipython->jupyter-dash>=0.4.2->plotly-resampler) (0.18.2)
Requirement already satisfied: pygments>=2.4.0 in ./conda/envs/work/lib/python3.10/site-packages (from ipython->jupyter-dash>=0.4.2->plotly-resampler) (2.14.0)
Requirement already satisfied: decorator in ./conda/envs/work/lib/python3.10/site-packages (from ipython->jupyter-dash>=0.4.2->plotly-resampler) (5.1.1)
Requirement already satisfied: charset-normalizer<4,>=2 in ./conda/envs/work/lib/python3.10/site-packages (from requests->jupyter-dash>=0.4.2->plotly-resampler) (2.1.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./conda/envs/work/lib/python3.10/site-packages (from requests->jupyter-dash>=0.4.2->plotly-resampler) (1.26.15)
Requirement already satisfied: idna<4,>=2.5 in ./conda/envs/work/lib/python3.10/site-packages (from requests->jupyter-dash>=0.4.2->plotly-resampler) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in ./conda/envs/work/lib/python3.10/site-packages (from requests->jupyter-dash>=0.4.2->plotly-resampler) (2022.12.7)
Requirement already satisfied: parso<0.9.0,>=0.8.0 in ./conda/envs/work/lib/python3.10/site-packages (from jedi>=0.16->ipython->jupyter-dash>=0.4.2->plotly-resampler) (0.8.3)
Requirement already satisfied: MarkupSafe>=2.0 in ./conda/envs/work/lib/python3.10/site-packages (from Jinja2>=3.0->Flask>=1.0.4->dash<3.0.0,>=2.2.0->plotly-resampler) (2.1.2)
Requirement already satisfied: platformdirs>=2.5 in ./conda/envs/work/lib/python3.10/site-packages (from jupyter-core!=5.0.*,>=4.12->ipykernel->jupyter-dash>=0.4.2->plotly-resampler) (3.2.0)
Requirement already satisfied: ptyprocess>=0.5 in ./conda/envs/work/lib/python3.10/site-packages (from pexpect>4.3->ipython->jupyter-dash>=0.4.2->plotly-resampler) (0.7.0)
Requirement already satisfied: wcwidth in ./conda/envs/work/lib/python3.10/site-packages (from prompt-toolkit!=3.0.37,<3.1.0,>=3.0.30->ipython->jupyter-dash>=0.4.2->plotly-resampler) (0.2.6)
Requirement already satisfied: pure-eval in ./conda/envs/work/lib/python3.10/site-packages (from stack-data->ipython->jupyter-dash>=0.4.2->plotly-resampler) (0.2.2)
Requirement already satisfied: executing>=1.2.0 in ./conda/envs/work/lib/python3.10/site-packages (from stack-data->ipython->jupyter-dash>=0.4.2->plotly-resampler) (1.2.0)
Requirement already satisfied: asttokens>=2.1.0 in ./conda/envs/work/lib/python3.10/site-packages (from stack-data->ipython->jupyter-dash>=0.4.2->plotly-resampler) (2.2.1)
Building wheels for collected packages: plotly-resampler
  Building wheel for plotly-resampler (pyproject.toml) ... done
  Created wheel for plotly-resampler: filename=plotly_resampler-0.8.3.2-cp310-cp310-macosx_12_0_arm64.whl size=51915 sha256=0a8881ef0d12384b58a9dff0fb06ec9416022285e6cb3dc24c92c92231c27957
  Stored in directory: /Users/white/Library/Caches/pip/wheels/a9/19/e2/ec856d98cf9b617511fff0abae1dfed5bf1f6093ce65126b45
Successfully built plotly-resampler
Installing collected packages: trace-updater, dash-table, dash-html-components, dash-core-components, retrying, orjson, ansi2html, jupyter-dash, plotly-resampler
Successfully installed ansi2html-1.8.0 dash-core-components-2.0.0 dash-html-components-2.0.0 dash-table-5.0.0 jupyter-dash-0.4.2 orjson-3.8.9 plotly-resampler-0.8.3.2 retrying-1.3.4 trace-updater-0.0.9
```

It largely leaned on my "daily driver" env, and that is primarily `conda-forge`. A couple of packages in there are `pypi`

**Comment by darynwhite** [comment]
I little more digging and backtracing and I believe it is tied to the instance that this library references is internal only. There is no attempt to look for an external `lttbc` instance, as seen here:

https://github.com/predict-idlab/plotly-resampler/blob/c60833eb4382b3072cb591b36de737dbf769a5ea/plotly_resampler/aggregation/aggregators.py#LL17-L26C62

```python
from ..aggregation.aggregation_interface import AbstractSeriesAggregator


try:
    # The efficient c version of the LTTB algorithm
    from .algorithms.lttb_c import LTTB_core_c as LTTB_core
except (ImportError, ModuleNotFoundError):
    import warnings


    warnings.warn("Could not import lttbc; will use a (slower) python alternative.")
    from .algorithms.lttb_py import LTTB_core_py as LTTB_core
```

**Comment by darynwhite** [comment]
Closed by #31
